lib/travis/logs/database.rb

```diff
@@ -205,6 +206,33 @@ def aggregatable_logs(regular_interval, force_interval, limit,
         query.map(:log_id).uniq
       end
 
+      def aggregatable_logs_by_partition(per_partition_limit,
+                                         order: :created_at,
+                                         cutoff: 1.day)
+        maint.restrict!
+        ids = []
+
+        cutoff = Time.now.utc - cutoff
+        cutoff_partition = cutoff.strftime('log_parts_p%Y_%m_%d')
+
+        partitions = db[
+          %[SELECT partman.show_partitions('public.log_parts')]
+        ].map do |row|
+          row.fetch(:show_partitions).gsub(/[()]/, '').split(',').last
```

Comment:
This makes me feel icky, but it's effectively the same thing that sequel's custom postgres type parsers do, so.... 🤷‍♀️